Add promote and demote

ItsMrReals opened the pull request:
- Added promote and demote
- Added promote and demoted stuff

Commit:
Update index.js

- Add promote
- Add demoted

Commit:
Update README.md

- Added promote docs
- Added demote docs

Commit:
Update index.js

Change GAR to GetRank

ChrisvanChip reviewed (changes requested):
The function name that could cause confusion (gar) was solved in a later commit, so that's fine. The error messages are not really descriptive and do not explain the issue (or contain grammatical issues). We would like to assist programmers before they have to contact us. 

`Groupid should be without the have no quotes` -> I don't get this sentence, could you rephrase this more clearly?
`Something went wrong ooof` -> It doesn't describe the issue and requires programmers to either contact us or debug the whole module by themselves.

Commit:
Update index.js

ItsMrReals commented:
Alrighty,
Issue code 0 was for testing, and the other code has been made more clear

Commit:
Update README.md

ChrisvanChip reviewed (changes requested):
Just for the little finishing touches, please use `console.error` or `reject` (when using promises) when something goes wrong. `console.log` might go unnoticed. `reject` can also be catched, which allows the programmer to make an error-proof system. 

After you've made those changes, it's up to @JonasTheGamer for making a final review. Thanks for contributing to RocketBots! 🚀

Commit:
Update index.js

ItsMrReals commented:
Done

Commit:
Update index.js

ChrisvanChip committed:
State credits at function level

Causes more possiblity to gain attention and allows others to contribute towards those functions by stating you've made the initial contribution.

ChrisvanChip reviewed (changes requested):
Take a look at your promotion function, line 88. Are you sure this intentional? It's performs the same as the demote function. And further, could you please remove the `console.log` you used for development? After that, it's perfect and you have my checkmark! ✅

Commit:
Update index.js

Fi

Commit:
Update index.js

C

Commit:
Update index.js

Fi

JonasTheGamer commented:
I will make an API endpoint on RocketApps for the getrank to have it go through our proxy.

JonasTheGamer commented:
@ItsMrReals Your code won't work in all usecases here. This is why:
Some groups have these roles: 
1: member
2: superior member
4: nicer member

However, if you are currently rank two and you do .promote() then the bot will attempt to rank 3 which doesn't exist, that'll error.

I would suggest writing the code as following:

function promote(userId)
     rank = getRank(userId)
     check if rank + 1 exists, if not, do +1 again.
     setRank(userId)
end

ChrisvanChip commented:
You could check out noblox.js' changeRank function:
https://github.com/suufi/noblox.js/blob/master/lib/group/changeRank.js
